examples/cookbook.cpp

```diff
@@ -68,24 +68,34 @@ GetTypeOfValue(JSContext* cx, JS::HandleValue v)
  * v = {};
  * v = new Symbol(someString);
  */
-static void
+static bool
 SetValue(JSContext* cx)
 {
   JS::RootedValue v(cx);
   JS::RootedString someString(cx, JS_NewStringCopyZ(cx, "my string"));
+  if (!someString)
+      return false;
   JS::RootedObject obj(cx, JS_NewPlainObject(cx));
+  if (!obj)
+      return false;
   JS::RootedSymbol symbol(cx, JS::NewSymbol(cx, someString));
+  if (!symbol)
+      return false;
 
   // clang-format off
   v.setInt32(0);           /* or: */ v = JS::Int32Value(0);
   v.setDouble(0.5);        /* or: */ v = JS::DoubleValue(0.5);
+  v.setNumber(0);          /* or: */ v = JS::NumberValue(0);
```

Comment:
`setNumber(0)` fails to compile for me, because clang can't decide between `setNumber(uint32_t)` or `setNumber(double)`. I'll fix it up by changing the argument to `0u`.